### PR TITLE
Added health visualisation

### PR DESCRIPTION
**Pull request by NickyW**
Health is displayed in numbers from 0-100
Sets an overlay to indicate health <= 50
Loops a sound when health <= 25

### CODE REVIEW

File: Confus/Confus/GUI.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	void GUI::drawBloodOverlay()
+	{
+		m_BloodOverlay->setVisible(m_PlayerNode->PlayerHealth.getHealth() <= 50);
+	}
+
+	void GUI::lowHealthAudio()
+	{
+		if (m_PlayerNode->PlayerHealth.getHealth() <= 25)
```
Comment (Athosvk) [inline Confus/Confus/GUI.cpp:45]: Please add curly brackets to indicate the scope of the if/else

File: Confus/Confus/Health.h [hunk 2]
```diff
@@ -10,6 +10,7 @@ namespace Confus
 		std::function<bool(irr::scene::ISceneNode* a_DamagedNode)> m_DeathCallback;
 		irr::scene::ISceneNode* m_AttachedNode;
 		int m_Health;
+		int m_MaxHealth = 100;
```
Comment (ChristianStiehl) [inline Confus/Confus/Health.h:13]: Should Health & MaxHealth really be a private member variable? I think we might want to access this in the future (power ups or health checks?). Would like a second opinion on this.

**Comment by ChristianStiehl** [comment]
Maybe find a better texture for the blood overlay, not very important, but not a lot of work either, could look a lot better with some actual blood drops or something a bit nicer looking.

File: Confus/Confus/Health.h (hunk 2 above)
Comment (Athosvk) [inline Confus/Confus/Health.h:13]: We can already change the health by using heal and damage, so I don't see the need for making it public. Can do for MaxHealth, but there's less harm in keeping it encapsulated like this. If needed in the future, you can always add a setter

File: Confus/Confus/GUI.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	{
+		m_GUIEnvironment = a_Device->getGUIEnvironment();
+
+		m_Driver = a_Device->getVideoDriver();
+		m_BloodImage = m_Driver->getTexture("Media/Textures/Blood.png");
+		m_BloodOverlay = m_GUIEnvironment->addImage(m_BloodImage, irr::core::position2d<int>(0, 0));
+		m_BloodOverlay->setVisible(false);
+		m_PlayerNode = a_Player;
+		m_HealthTextBox = m_GUIEnvironment->addStaticText(L"", irr::core::rect<irr::s32>(10, 10, 100, 25), false);
+		m_AudioSourceLowHealth = new OpenALSource("heartbeat.wav");
```
Comment (Lansenou) [inline Confus/Confus/GUI.cpp:17]: Memory leak.

File: Confus/Confus/GUI.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	void GUI::update()
+	{
+		m_HealthString = "";
+		m_HealthString += L"Health: ";
+		m_HealthString += m_PlayerNode->PlayerHealth.getHealth();
+		m_HealthString += L"%";
```
Comment (Lansenou) [inline Confus/Confus/GUI.cpp]: Not important but, what about:
`m_HealthString = (L"Health: " + std::to_wstring(m_PlayerNode->PlayerHealth.getHealth()) + L"%").c_str();
 m_HealthTextBox->setText(m_HealthString.c_str());`

Comment (Athosvk) [inline Confus/Confus/GUI.cpp]: Yeah seems a bit more to the point, though the outer most parenthesis aren't really necessary

File: Confus/Confus/GUI.cpp (hunk 3 above)
Comment (NickyW) [inline Confus/Confus/GUI.cpp:17]: Any suggestions?
PlayerAudioEmitter.cpp seems to do this in the same way, on a side note.

Comment (Athosvk) [inline Confus/Confus/GUI.cpp:17]: Yes, but it deallocates the memory as well. Take a good look at its destructor for the memory it has dynamically allocated

**Comment by Lansenou** [comment]
Please fix merge conflicts.

**Comment by NickyW** [comment]
Closed, made a new one.

**Pull request closed without merging**